dbtc/client/cloud/base.py

```diff
@@ -1002,6 +1035,279 @@ def test_connection(self, account_id: int, payload: Dict) -> Dict:
+                Remove a job replicated via autoscaling after it finishes running.
+            autoscale_job_identifier (str, optional): Only relevant when job_run_strategy = 'autoscale'
+                append value to the existing job name when replicating the job definition.
+                If None defaults to the current timestamp on job creation
+        """
+        
+        is_new_job_created = False 
+
+        self.console.log(
+            'Triggered with autoscaling set to True. '
+            'Detecting any running instances'
+        )
+        most_recent_job_run = self.list_runs(
+            account_id=account_id, job_definition_id=job_id, limit=1, order_by='-id'
+        )['data'][0]
+        most_recent_job_run_status = most_recent_job_run['status_humanized']
+
+        self.console.log(
+            f'Status for most recent run of job {job_id} '
+            f'is {most_recent_job_run_status}.'
+        )
+
+        if most_recent_job_run_status not in ['Queued', 'Starting', 'Running']:
+            self.console.log(
+                f'autoscale set to true but base job with id {job_id} is free. '
+                'triggering base job and ignoring autoscale configuration.'
+            )
+
+        else:
+            self.console.log(f'job_id {job_id} has an active run. Cloning job.')
+            optional_fields = [k for k,v in models.Job.__fields__.items() if not v.required]
+            required_fields = [k for k,v in models.Job.__fields__.items() if v.required]
+            existing_job_defintion = self.get_job(
+                account_id=account_id, job_id=job_id
+            )['data']
+
+            new_job_definition = {k:v for k,v in existing_job_defintion.items() if k in required_fields}
+            
+            if not autoscale_job_identifier:
+                creation_time = datetime.now().strftime('%Y-%m-%d-%H-%M-%S')
+                new_job_name = '-'.join([new_job_definition['name'], creation_time])
+                new_job_definition['name'] = new_job_name
+            else:
+                new_job_name = '-'.join([new_job_definition['name'], autoscale_job_identifier])
+                new_job_definition['name'] = new_job_name
+            
+            # make sure the id is none on the new job
+            new_job_definition['id'] = None
+            job_id = self.create_job(
+                account_id=account_id, payload=new_job_definition
+            )['data']['id']
+
+            # fill in optional fields on the job if present in the source
+            for field in optional_fields:
+                new_job_definition[field] = existing_job_defintion.get(field)
+            
+            # set the ID in the update request
+            new_job_definition['id'] = job_id
+            self.console.log(new_job_definition)
+            job_id = self.update_job(
+                account_id=account_id, job_id=job_id, payload=new_job_definition
+            )['data']['id']
```

Comment:
@matt-winkler I'm not sure I follow here.  Why wouldn't we just create the new job from the existing job definition (outside of setting `id` to `None` and changing the `name`)?

Reply:
@dpguthrie we can't copy the existing data in the job 1:1 in a new CREATE JOB request. For example, you can't create a new job including the `deferring_job_definition_id` parameter.

Reply:
@dpguthrie updated my comment with a better example